### DISCUSSION TITLE
Does anyone have a dependable way of extracting selected text from any app?

### DISCUSSION THREAD

**Discussion by Madd0g** [Hammerspoon/hammerspoon#3361] [General]
I'm using `hs.axuielement`, I found the `AXSelectedText` attribute, it seems like in chromium/electron apps the selected text comes out without the new lines.

I tried playing with ranges to extract the text from `AXValue`, but the ranges all wrong and the text offset comes out all wrong.

Does anyone have a good implementation that extracts texts properly with `hs.axuielement`?

Thanks!

**Comment by boblauer** [comment]
I know this is old, but I wonder if you could, after a text selection is detected:

- Save what's currently in the clipboard to $variable
- Send cmd+c to copy the selected text
- Read it from the clipboard
- Insert $variable back into the clipboard
